api/policies.go

```diff
@@ -13,105 +13,95 @@ import (
+}
 
-	if _, err := writer.Write(b); err != nil {
-		log.Error(ctx, "getPolicy Handler: error writing bytes to response", err, logData)
-		http.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+func handleGetPolicyError(ctx context.Context, err error, policyId string) *models.ErrorResponse {
+	logData := log.Data{"policy_id": policyId}
+	if err == apierrors.ErrPolicyNotFound {
+		return models.NewErrorResponse(http.StatusNotFound,
+			nil,
+			models.NewError(ctx, err, models.PolicyNotFoundError, models.PolicyNotFoundDescription, logData),
+		)
 	}
-	log.Info(ctx, "getPolicy Handler: Successfully retrieved policy", logData)
+	return models.NewErrorResponse(http.StatusInternalServerError,
+		nil,
+		models.NewError(ctx, err, models.GetPolicyError, models.GetPolicyErrorDescription, logData),
+	)
 }
 
 //DeletePolicyHandler is a handler that deletes policy by its ID from DB
-func (api *API) DeletePolicyHandler(writer http.ResponseWriter, request *http.Request) {
-
-	ctx := request.Context()
-	vars := mux.Vars(request)
+func (api *API) DeletePolicyHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	vars := mux.Vars(req)
 	policyId := vars["id"]
-	logData := log.Data{"policy_id": policyId}
 
 	err := api.permissionsStore.DeletePolicy(ctx, policyId)
 	if err != nil {
-		log.Error(ctx, "deletePolicy Handler: deleting policy in DB returned an error", err, logData)
-		if err == apierrors.ErrPolicyNotFound {
-			http.Error(writer, err.Error(), http.StatusNotFound)
-			return
-		}
-		http.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+		return nil, handleDeletePolicyError(ctx, err, policyId)
 	}
 
-	writer.WriteHeader(http.StatusNoContent)
-	log.Info(ctx, "deletePolicy Handler: Successfully deleted policy", logData)
+	return models.NewSuccessResponse(nil, http.StatusNoContent, nil), nil
 }
 
-//PostPolicyHandler is a handler that creates a new policies in DB
-func (api *API) PostPolicyHandler(writer http.ResponseWriter, request *http.Request) {
+func handleDeletePolicyError(ctx context.Context, err error, policyId string) *models.ErrorResponse {
+	logData := log.Data{"policy_id": policyId}
+	if err == apierrors.ErrPolicyNotFound {
+		return models.NewErrorResponse(http.StatusNotFound,
+			nil,
+			models.NewError(ctx, err, models.PolicyNotFoundError, models.PolicyNotFoundDescription, logData),
+		)
+	}
+	return models.NewErrorResponse(http.StatusInternalServerError,
+		nil,
+		models.NewError(ctx, err, models.DeletePolicyError, models.DeletePolicyErrorDescription, logData),
+	)
+}
 
-	ctx := request.Context()
-	policy, err := models.CreatePolicy(request.Body)
+//PostPolicyHandler is a handler that creates a new policies in DB
+func (api *API) PostPolicyHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	policy, err := models.CreatePolicy(req.Body)
```

Comment:
Im not 100% sure about this but I think we might have to do `defer req.Body.close` to ensure we're cleaning up properly.

Reply:
Nope, I  believe we good https://github.com/golang/go/blob/master/src/net/http/request.go#L174